src/message.rs

```diff
@@ -0,0 +1,256 @@
+use amqp_worker::{
+  job::{Job, JobResult, JobStatus},
+  MessageError, ParametersContainer,
+};
+use lapin_futures::Channel;
+use stainless_ffmpeg::format_context::FormatContext;
+
+const SEGMENT_DURATION_PARAMETER: &'static str = "segment_duration";
+const SEGMENT_OVERLAP_PARAMETER: &'static str = "segment_overlap";
+const SOURCE_PATH_PARAMETER: &'static str = "source_path";
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Segment {
+  start: u64,
+  end: u64,
+}
+
+impl Segment {
+  fn new(start: u64, end: u64) -> Segment {
+    Segment { start, end }
+  }
+}
+
+pub fn process(
+  _channel: Option<&Channel>,
+  job: &Job,
+  job_result: JobResult,
+) -> Result<JobResult, MessageError> {
+  let source_path = job
+    .get_string_parameter(SOURCE_PATH_PARAMETER)
+    .ok_or_else(|| {
+      MessageError::ProcessingError(
+        job_result
+          .clone()
+          .with_status(JobStatus::Error)
+          .with_message(&format!(
+            "Invalid job message: missing expected '{}' parameter.",
+            SEGMENT_DURATION_PARAMETER
+          )),
+      )
+    })?;
+
+  let segment_duration = job
+    .get_integer_parameter(SEGMENT_DURATION_PARAMETER)
+    .ok_or_else(|| {
+      MessageError::ProcessingError(
+        job_result
+          .clone()
+          .with_status(JobStatus::Error)
+          .with_message(&format!(
+            "Invalid job message: missing expected '{}' parameter.",
+            SEGMENT_DURATION_PARAMETER
+          )),
+      )
+    })?;
+
+  let segment_overlap = job.get_integer_parameter(SEGMENT_OVERLAP_PARAMETER);
+
+  let media_duration = get_media_duration_in_milliseconds(&source_path).map_err(|msg| {
+    MessageError::ProcessingError(
+      job_result
+        .clone()
+        .with_status(JobStatus::Error)
+        .with_message(&msg),
+    )
+  })?;
+
+  debug!("Input media duration: {} ms", media_duration);
+
+  let segments = split_media(media_duration, segment_duration, segment_overlap).map_err(|msg| {
+    MessageError::ProcessingError(
+      job_result
+        .clone()
+        .with_status(JobStatus::Error)
+        .with_message(&msg),
+    )
+  })?;
+
+  let result = serde_json::to_string(&segments).map_err(|error| {
+    MessageError::ProcessingError(
+      job_result
+        .clone()
+        .with_status(JobStatus::Error)
+        .with_message(&format!("{}", error)),
+    )
+  })?;
+
+  Ok(
+    job_result
+      .with_status(JobStatus::Completed)
+      .with_message(&result),
+  )
+}
+
+fn get_media_duration_in_milliseconds(path: &str) -> Result<u64, String> {
+  let mut format_context = FormatContext::new(path)?;
+  format_context.open_input()?;
+
+  let duration_millisec = if let Some(duration) = format_context.get_duration() {
+    (duration * 1000.) as u64
+  } else {
+    0
+  };
+
+  format_context.close_input();
+  Ok(duration_millisec)
+}
+
+fn split_media(
+  media_duration: u64,
+  segment_duration: i64,
```

Comment:
Can you add also a parameter to mention: `Split in X number of parts` without take care of the duration of each segment.

So SegmentDuration and NumberOfParts can be wrapped into an Enum

Reply:
See  318bdef